SlideController/Classes/SlideContentView.swift

```diff
@@ -82,52 +85,48 @@ extension ViewSlidableImplementation: ViewSlidable {
         }
-        
         for i in 0...views.count - 1 {
             let view = views[i]
             view.backgroundColor = UIColor.clear
             view.translatesAutoresizingMaskIntoConstraints = false
             let viewModel = SlideContainerView(view: view)
-            pages.append(viewModel)
+            containers.append(viewModel)
             addSubview(view)
-            activateConstraints(viewModel, prevPage: prevPage, isLast: i == views.count - 1, direction: scrollDirection)
+            activateConstraints(viewModel, prevPage: prevPage, isLast: i == views.count - 1, direction: slideDirection)
             prevPage = viewModel
         }
     }
     
     func insertView(view: View, index: Int) {
-        guard index < pages.count else { return }
+        guard index < containers.count else { return }
         view.backgroundColor = UIColor.clear
         view.translatesAutoresizingMaskIntoConstraints = false
         let viewModel = SlideContainerView(view: view)
-        pages.insert(viewModel, at: index)
+        containers.insert(viewModel, at: index)
         addSubview(view)
-        let prevPage: SlideContainerView? = index > 0 ? pages[index - 1]:  nil
-        let nextPage: SlideContainerView = pages[index + 1]
-        activateConstraints(viewModel, prevPage: prevPage, isLast: false, direction: scrollDirection)
-        updateConstraints(nextPage, prevPage: viewModel, isLast: index == pages.count - 2, direction: scrollDirection)
+        let prevPage: SlideContainerView? = index > 0 ? containers[index - 1]:  nil
+        let nextPage: SlideContainerView = containers[index + 1]
+        activateConstraints(viewModel, prevPage: prevPage, isLast: false, direction: slideDirection)
+        updateConstraints(nextPage, prevPage: viewModel, isLast: index == containers.count - 2, direction: slideDirection)
     }
     
     func removeViewAtIndex(index: Int) {
-        guard index < pages.count else { return }
-        let page: SlideContainerView = pages[index]
-        let prevPage: SlideContainerView? = index > 0 ? pages[index - 1]: nil
-        let nextPage: SlideContainerView? = index < pages.count - 1 ? pages[index + 1]: nil
-        
-        pages.remove(at: index)
+        guard index < containers.count else { return }
+        let page: SlideContainerView = containers[index]
+        let prevPage: SlideContainerView? = index > 0 ? containers[index - 1]: nil
+        let nextPage: SlideContainerView? = index < containers.count - 1 ? containers[index + 1]: nil
+        containers.remove(at: index)
         page.view.removeFromSuperview()
         if let nextPage = nextPage {
-            updateConstraints(nextPage, prevPage: prevPage, isLast: index == pages.count - 1, direction: scrollDirection)
+            updateConstraints(nextPage, prevPage: prevPage, isLast: index == containers.count - 1, direction: slideDirection)
```

Comment:
`isLast` will be always false if nextPage is not nil.

Reply:
not for insertView case. @oroom you can refactor this yourself, but fix requires some thinking time :)